### ISSUE TITLE
M2 reader not connecting on 3.3.X with firmware version 2.01.00.15 while latest firmware 2.01.00.20 connects

### ISSUE THREAD

**Issue by ashare80** [stripe/stripe-terminal-android#430]
## Summary
Received new M2 devices and they hang on calls to connect via Bluetooth and USB. After `onConnectionStatusChange` event of "CONNECTING" there are no callbacks to `ReaderCallback` or `ReaderListener`. Only after disconnecting the reader by turning off or unplugging USB does the error callback get called with the exception.

The last logs we see before it hangs an no more logs or callbacks occur are:
```
StripeTerminal          E  class=BbposDeviceListenerWrapperImpl message="Failed to read terminal setting" DF8926=TAG_NOT_ALLOWED_TO_ACCESS
StripeTerminal          E  class=BbposDeviceListenerWrapperImpl message="Failed to read terminal setting" DF8438=TAG_NOT_FOUND
StripeTerminal          I  class=BbposControllerListener message=onReturnReadTerminalSettingResult
```

Downgrading to `3.2.1` resolves and the reader correctly connects and starts the required reader update

## Android version
Android 11

## Impacted devices (Android devices or readers)
M2

## SDK version
 3.3.0 and  3.3.1

**Comment by jbring17** [comment]
We are experiencing something very similar. For an out of the box reader, we can connect and start the update on 3.2.1, but on an upgrade to 3.3.0 and 3.3.1, we get stuck on connecting.

**Comment by ashare80** [comment]
> @ashare80 thanks for raising this, we've identified the issue and would be releasing a fix soon

Thanks!!

**Comment by ugochukwu-stripe** [comment]
This issue should now be resolved by SDK 3.4.0
